phoenix-core/src/it/java/org/apache/phoenix/end2end/MutationBatchFailedStateMetricIT.java

```diff
@@ -0,0 +1,212 @@
+    String create_index = "CREATE INDEX " + indexName + " ON %s(VAL1 DESC) INCLUDE (VAL2)";
+    String upsertStatement = "UPSERT INTO %s VALUES(?, ?, ?)";
+    String deleteTableName = generateUniqueName();
+    private final boolean transactional;
+
+    public MutationBatchFailedStateMetricIT(String transactionProvider) {
+        this.transactional = transactionProvider != null;
+        if (this.transactional) {
+            create_table =
+                    create_table + (this.transactional
+                            ? (" TRANSACTIONAL=true,TRANSACTION_PROVIDER='" + transactionProvider
+                                    + "'")
+                            : "");
+        }
+        createTables();
+        populateTables();
+    }
+
+    @BeforeClass
+    public static synchronized void doSetup() throws Exception {
+        Map<String, String> serverProps = Maps.newHashMapWithExpectedSize(3);
+        serverProps.put("hbase.coprocessor.abortonerror", "false");
+        serverProps.put(Indexer.CHECK_VERSION_CONF_KEY, "false");
+        Map<String, String> clientProps = Maps.newHashMapWithExpectedSize(2);
+        clientProps.put(QueryServices.COLLECT_REQUEST_LEVEL_METRICS, String.valueOf(true));
+        clientProps.put(QueryServices.TRANSACTIONS_ENABLED, "true");
+        setUpTestDriver(new ReadOnlyProps(serverProps.entrySet().iterator()),
+            new ReadOnlyProps(clientProps.entrySet().iterator()));
+
+    }
+
+    @Parameterized.Parameters(name = "MutationBatchFailedStateMetricIT_transactionProvider={0}")
+    public static synchronized Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] { { "OMID" }, { null } });
+    }
+
+    /**
+     * We will have 5 rows in table and we will try deleting 4 out of 5 rows
+     * In this test intention is to fail the delete and see how metric value is updated when that happens
+     * for transactional table and non transactional table
+     * @throws SQLException
+     */
+    @Test
+    public void testFailedDelete() throws Exception {
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        Connection conn = null;
+        try {
+            conn = DriverManager.getConnection(getUrl(), props);
+
+            Statement stmt = conn.createStatement();
+            // adding coprocessor which basically overrides prebatchmutate whiich is called for all
+            // the mutations
+            // it is called before applying mutation to region
+            TestUtil.addCoprocessor(conn, deleteTableName,
+                ImmutableIndexIT.DeleteFailingRegionObserver.class);
+            // trying to delete 4 rows with this single delete statement
+            String dml = String.format("DELETE FROM %s where val1 >  1", deleteTableName);
+            stmt.execute(dml);
+            conn.commit();
+
+            TestUtil.removeCoprocessor(conn, deleteTableName,
+                ImmutableIndexIT.DeleteFailingRegionObserver.class);
```

Comment:
We can remove this as it will not likely be executed for successful test scenario.